### PR TITLE
Refactor tags to skill

### PR DESCRIPTION
**Pull request by goel-a**
- Refactor code in Storage, Ui, model and Command to reflect change from tag to skill
- Update add command to accept s/ prefix for skills
- Update test cases to show change to skills
- Minor update to UI to reflect UI mockup
#52 
#49

### CODE REVIEW

File: src/main/java/seedu/linkedout/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -101,24 +101,24 @@ public static Email parseEmail(String email) throws ParseException {
...
      * @throws ParseException if the given {@code tag} is invalid.
      */
-    public static Tag parseTag(String tag) throws ParseException {
-        requireNonNull(tag);
-        String trimmedTag = tag.trim();
-        if (!Tag.isValidTagName(trimmedTag)) {
-            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+    public static Skill parseSkill(String skill) throws ParseException {
+        requireNonNull(skill);
+        String trimmedTag = skill.trim();
```
Comment (RyanCheungJF) [inline src/main/java/seedu/linkedout/logic/parser/ParserUtil.java:106]: Change variable name to reflect our app as well.
```suggestion
        String trimmedSkill = skill.trim();
```

File: src/main/java/seedu/linkedout/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -101,24 +101,24 @@ public static Email parseEmail(String email) throws ParseException {
...
      */
-    public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
-        requireNonNull(tags);
-        final Set<Tag> tagSet = new HashSet<>();
-        for (String tagName : tags) {
-            tagSet.add(parseTag(tagName));
+    public static Set<Skill> parseSkills(Collection<String> skills) throws ParseException {
+        requireNonNull(skills);
+        final Set<Skill> skillSet = new HashSet<>();
+        for (String tagName : skills) {
```
Comment (RyanCheungJF) [inline src/main/java/seedu/linkedout/logic/parser/ParserUtil.java:119]:
```suggestion
        for (String skillName : skills) {
```

File: src/test/java/seedu/linkedout/testutil/ApplicantBuilder.java [hunk 3]
```diff
@@ -60,8 +60,8 @@ public ApplicantBuilder withName(String name) {
     /**
      * Parses the {@code tags} into a {@code Set<Tag>} and set it to the {@code Applicant} that we are building.
```
Comment (RyanCheungJF) [inline src/test/java/seedu/linkedout/testutil/ApplicantBuilder.java:61]:
```suggestion
     * Parses the {@code skills} into a {@code Set<Skill>} and set it to the {@code Applicant} that we are building.
```

**Review by RyanCheungJF — COMMENTED** [review]
Good refactor, just a few small changes to naming that need to be changed. Nice UI change!

**Review by kaiyichen — COMMENTED** [review]
Looks good overall. Minor changes : some method header needed to be changed to "skills" instead of "tags"

**Review by kaiyichen — APPROVED** [review]
LGTM!!

**Pull request merged**